lib/omnibus/packagers/windows_base.rb

```diff
@@ -141,17 +203,30 @@ def sign_package(package_file, is_bundle: false )
     end
 
     def try_sign(package_file, url)
-      cmd = Array.new.tap do |arr|
-        arr << "signtool.exe"
-        arr << "sign /v"
-        arr << "/t #{url}"
-        arr << "/fd #{algorithm}"
-        arr << "/sm" if machine_store?
-        arr << "/s #{cert_store_name}"
-        arr << "/sha1 #{thumbprint}"
-        arr << "/d #{project.package_name}"
-        arr << "\"#{package_file}\""
-      end.join(" ")
+      if signing_identity
+        cmd = Array.new.tap do |arr|
+          arr << "signtool.exe"
+          arr << "sign /v"
+          arr << "/t #{url}"
+          arr << "/fd #{algorithm}"
+          arr << "/sm" if machine_store?
+          arr << "/s #{cert_store_name}"
+          arr << "/sha1 #{thumbprint}"
+          arr << "/d #{project.package_name}"
+          arr << "\"#{package_file}\""
+        end.join(" ")
+      elsif signing_identity_file
+        cmd = Array.new.tap do |arr|
+          arr << "signtool.exe"
+          arr << "sign /v"
+          arr << "/t #{url}"
+          arr << "/f #{pfx_file}"
+          arr << "/fd #{algorithm}"
```

Comment:
should this be `algorithm` so that if `signing_identity` is defined it takes precedence over whatever is defined in the `signing_identity_file`? Or should this maybe better be `pfx_algorithm`?

Reply:
I don't understand the question. `algorithm` is the algorithm used to compute the digest for the file, it's independent of how we get the key (via file or stored in the registry)

`
/fd | Specifies the file digest algorithm to use to create file signatures. The default algorithm is Secure Hash Algorithm (SHA-1).Windows Vista and earlier: This flag is not supported.
`

Reply:
This is related to the mutual exclusive comment I made earlier:

The code looks like this:
```
 def algorithm
      if signing_identity
        signing_identity[:algorithm]
      elsif signing_identity_file
        signing_identity_file[:algorithm]
      else
        nil
      end
    end
```

So it would take precedence of what might be in the file... (ie. the algorithms could be different). So wondering if it wouldn't be safer to just use `pfx_algorithm` here instead. Or if we implement an exception if they're both defined, then this comment would be mute.

Reply:
I just made them mutually exclusive.